api/app/Builders/PoolCandidateBuilder.php

```diff
@@ -0,0 +1,565 @@
+    {
+        if (empty($classifications)) {
+            return $this;
+        }
+
+        return $this->whereAvailable()
+            ->whereAppliedClassificationsIn($classifications);
+    }
+
+    /**
+     * Scope Publishing Groups
+     *
+     * Restrict a query by specific publishing groups
+     */
+    public function wherePublishingGroupsIn(?array $publishingGroups): self
+    {
+        if (empty($publishingGroups)) {
+            return $this;
+        }
+
+        return $this->whereHas('pool', function (Builder $query) use ($publishingGroups) {
+            /** @var \App\Builders\PoolBuilder $query */
+            $query->publishingGroups($publishingGroups);
+        });
+    }
+
+    /**
+     * Scope is IT & OTHER Publishing Groups
+     *
+     * Restrict a query by pool candidates that are for pools
+     * containing IT and OTHER publishing groups
+     */
+    public function whereInTalentSearchablePublishingGroup(): self
+    {
+        return $this->wherePublishingGroupsIn([
+            PublishingGroup::IT_JOBS->name,
+            PublishingGroup::OTHER->name,
+        ]);
+
+    }
+
+    public function whereOperationalRequirementsIn(?array $operationalRequirements): self
+    {
+        if (empty($operationalRequirements)) {
+            return $this;
+        }
+
+        return $this->whereHas('user', function ($query) use ($operationalRequirements) {
+            $query->whereOperationalRequirementsIn($operationalRequirements);
+        });
+    }
+
+    public function whereLocationPreferencesIn(?array $workRegions): self
+    {
+        if (empty($workRegions)) {
+            return $this;
+        }
+
+        return $this->whereHas('user', function ($query) use ($workRegions) {
+            $query->whereLocationPreferencesIn($workRegions);
+        });
+    }
+
+    public function whereLanguageAbility(?string $languageAbility): self
+    {
+        if (empty($languageAbility)) {
+            return $this;
+        }
+
+        return $this->whereHas('user', function ($query) use ($languageAbility) {
+            $query->whereLanguageAbility($languageAbility);
+        });
+    }
+
+    public function whereAvailableInPools(?array $poolIds): self
+    {
+        if (empty($poolIds)) {
+            return $this;
+        }
+
+        return $this->whereIn('pool_id', $poolIds);
+    }
+
+    public function whereEquityIn(?array $equity): self
+    {
+        if (empty($equity)) {
+            return $this;
+        }
+
+        return $this->whereHas('user', function ($query) use ($equity) {
+            $query->whereEquityIn($equity);
+        });
+    }
+
+    public function whereGeneralSearch(?string $searchTerm): self
+    {
+        if (empty($searchTerm)) {
+            return $this;
+        }
+
+        return $this->whereHas('user', function ($userQuery) use ($searchTerm) {
+            $userQuery->whereGeneralSearch($searchTerm);
```

Comment:
Scrolling through and I see a phase of using `$userQuery`, an alternative to the above and neat way to name the variable off the builder in play